### PR TITLE
Added more env variables + npm scripts for local setup + fallback when mailgun key is missing

### PR DESCRIPTION
**Pull request by FabianNowak**
Improved the environment setup, so it is easier to completely run locally.

Replaced some hardcoded URLs/domains with environment variables.

New environment variables:

- MAILGUN_HOST (in case a Mailgun account runs on api.mailgun.net instead of api.eu.mailgun.net)
- API_URL (was "https://api.genecruncher.com", can now be set to localhost etc.)
- FRONTEND_URL (was "https://genecruncher.com", can now be set to localhost etc.)

Added a docker-compose file which starts a minio s3 instance and a mongodb instance
Added a npm script which

1. starts the docker compose for the s3 and mongodb and
2. starts the backend server without docker, using nodemon (added by @Pellumbengineer)

### CODE REVIEW

**Comment by FabianNowak** [comment]
**Additional commit:**
When no mailgun API key is specified, the email content will now instead be printed to the console, so a local test environment can run without a connection to mailgun. 
This way, one can access verification tokens or any other future tokens sent by email without ...
1. needing a mailgun key
2. using a real e-mail address

File: Backend/package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "start": "node ./build/server.js",
     "gcp-build": "tsc -p .",
     "dev:start": "npm-run-all gcp-build start",
-    "dev": "nodemon --watch src -e ts,ejs --exec npm run dev:start"
+    "dev": "nodemon --watch src -e ts,ejs --exec npm run dev:start",
+    "full-backend-local:docker": "docker-compose -f dev/docker-compose.yml up -d",
+    "full-backend-local": "npm-run-all full-backend-local:docker dev"
```
Comment (Pellumbengineer) [inline Backend/package.json:13]: After the full start of docker compose we still start "dev" right? I was wondering what is the reasoning about that?

Comment (FabianNowak) [reply Backend/package.json:13]: The new docker-compose file, that I added, only starts the database + s3, but not the backend server. The backend api is started separately. This way, the backend server can watch code file changes and simply connects to the dockerized db+s3.

Comment (Pellumbengineer) [reply Backend/package.json:13]: Ah okay!

**Review by Pellumbengineer — APPROVED** [review]

**Pull request merged**